src/pudl/analysis/timeseries_cleaning.py

```diff
@@ -1530,63 +1586,6 @@ def impute_flagged_values(
     return pd.concat(results)
 
 
-@pa.check_types
```

Comment:
Previously, this method did two things:
1. Find respondent/year combos that are missing lots of data and drop those years
2. Drop any respondents entirely that don't have any data across all years

I've split this into a new `flag_bad_years` function that does `1.`, plus properly assigns flags to all values in these years, and dropping blank respondents is now just moved into the top level `_flag_timeseries_matrix` function.